packages/ai/src/ai/modules/task/test_temp_file_security.py

```
@@ -0,0 +1,141 @@
+        # Bind the real method to our mock
+        mock._write_task_file = Task._write_task_file.__get__(mock, Task)
+        return mock
+
+    def test_file_permissions_owner_only(self):
+        """Temporary file must be created with 0o600 (owner read/write only)."""
+        engine = self._make_task_engine()
+        taskpath = asyncio.run(engine._write_task_file())
+        try:
+            file_stat = os.stat(taskpath)
+            mode = stat.S_IMODE(file_stat.st_mode)
+            self.assertEqual(
```

Ruff (0.15.6)

[warning] Use a regular `assert` instead of unittest-style `assertEqual`

Replace `assertEqual(...)` with `assert ...`

(PT009)